gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -193,3 +194,86 @@ def __init__(
 
   def expand(self, pvalue):
     return pvalue.pipeline | Read(self._source)
+
+
+class _WriteVcfHeaderFn(beam.DoFn):
+  """A DoFn for writing VCF headers to a file."""
+
+  HEADER_TEMPLATE = '##{}=<{}>\n'
+  FINAL_HEADER_LINE = '#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT\n'
+  INFO_KEY = 'INFO'
+  FILTER_KEY = 'FILTER'
+  ALT_KEY = 'ALT'
+  FORMAT_KEY = 'FORMAT'
+  CONTIG_KEY = 'contig'
+  ID_KEY = 'ID'
+  NUMBER_KEY = 'Number'
+  TYPE_KEY = 'Type'
+  DESCRIPTION_KEY = 'Description'
+
+  def __init__(self, file_path):
+    self._file_path = file_path
+
+  def process(self, header):
+    with FileSystems.create(self._file_path) as file_to_write:
+      self._write_headers_by_key(file_to_write, self.INFO_KEY, header.infos)
+      self._write_headers_by_key(file_to_write, self.FILTER_KEY, header.filters)
+      self._write_headers_by_key(file_to_write, self.ALT_KEY, header.alts)
+      self._write_headers_by_key(file_to_write, self.FORMAT_KEY, header.formats)
+      self._write_headers_by_key(file_to_write, self.CONTIG_KEY, header.contigs)
+      file_to_write.write(self.FINAL_HEADER_LINE)
+
+  def _write_headers_by_key(self, file_to_write, header_key, header):
+    if header_key == self.INFO_KEY:
+      self._remove_source_and_version(header)
+
+    for header_value in header.values():
+      file_to_write.write(self._to_vcf_header_line(header_key, header_value))
+
+  def _to_vcf_header_line(self, header_key, header):
+    headers = ['{}={}'.format(*self._map_header(header, key)) for key in header]
+    return self.HEADER_TEMPLATE.format(header_key, ','.join(headers))
+
+  def _map_header(self, header, key):
+    value = header[key]
+    key = self._map_key(key)
+    if key == self.NUMBER_KEY:
+      value = self._map_num(value)
+    elif key == self.DESCRIPTION_KEY:
+      value = '"{}"'.format(value)
+    return (key, value)
+
+  def _map_key(self, key):
```

Comment:
Please rename to something more specific...._get_formatted_header_key?

Reply:
Done.